programs/survival/auto_lighter.sc

```diff
@@ -12,11 +12,12 @@ __on_player_uses_item(player, item, hand) ->
 		ench = item:2:'Enchantments[]';
 		global_spread_love = 0;
 		delete(item:2:'Enchantments');
-		if (!ench,
+		if (!ench && player~'gamemode_id'!=3,
 			global_spread_love = 1;
 			if (ench==null, item:2 = nbt('{}'));
 			put(item:2:'Enchantments','[]');
 			put(item:2:'Enchantments', '{lvl:1s,id:"minecraft:protection"}', 0);
+		    	global_survival=!(player~'gamemode_id' % 2);
 			schedule(0, 'spread_torches', player);
```

Comment:
man, why globals, just add initial gamemode as an arguement to spread_torches.

Reply:
ok, man, why u bi h8'in on globalization? It's not like it caused a global pandemic or anything!